process/transactionProcessor.go

```diff
@@ -321,23 +321,37 @@ func (tp *TransactionProcessor) GetTransactionStatus(txHash string, sender strin
 	}
 
 	// get status of transaction from random observers
-	tx, err := tp.getTxFromObservers(txHash)
+	tx, err := tp.getTxFromObservers(txHash, false)
 	if err != nil {
 		return UnknownStatusTx, errors.ErrTransactionNotFound
 	}
 
 	return string(tx.Status), nil
 }
 
-func (tp *TransactionProcessor) getTxFromObservers(txHash string) (*data.FullTransaction, error) {
-	allObservers, err := tp.getObserversOrFullHistoryNodes()
+func (tp *TransactionProcessor) getTxFromObservers(txHash string, withFullHistoryNodes bool) (*data.FullTransaction, error) {
+	allObservers, err := tp.getObserversOrFullHistoryNodes(withFullHistoryNodes)
```

Comment:
Why are you returning the list of the observers that will need to be queried if you actually do not use them? (only their shard ID)

Reply:
done. replaced with a get shard ids function